doc/workshops/scientific-computing.rst

```diff
@@ -2,4 +2,730 @@
+
+.. note::
+
+   The *null hypothesis* is an alternative prediction to the one we have stated in our hypothesis. The null hypothesis simply states that the observed differences in means are due to random variations that occur when samples are collected.
+
+An ANOVA produces an *F-ratio* (also called F-statistic) that compares the amount of systematic variation in the data (variation that can be explained by the model or hypothesis) with the amount of unsystematic variance (variation that cannot be explained by our hypothesis or model). This means that F is the ratio of the model to its error. ANOVA also produces a *p-value* that indicates the probability that the variation can be attributed to the null hypothesis. The smaller the p-value, the less likely it is that the observed variation is due to chance.
+
+To perform an ANOVA, let's start by pivoting the dataset with the cognitive ability test results. The religiosity values will now be the column names, and each row will represent a person.
+
+.. code-block:: python
+
+   relig_pivot = ca_test.pivot(columns="q41", values="cognitive_score")
+
+You will see that one of the columns includes the answers to the test made by people who did not specify their feelings towards religion. Let's delete that column.
+
+.. code-block:: python
+
+   relig_pivot.drop(relig_pivot.columns[0], axis=1, inplace=True)
+
+.. image:: images/workshop-2/religiosity-pivot-table.png
+   :alt: Religiosity pivot table for ANOVA analysis.
+
+Now let's rename the column names (for simplicity) and remove the records that have missing values (with ``dropna``). The argument ``inplace=True`` allows you to directly modify a dataframe without having to create a new one.
+
+.. code-block:: python
+
+   relig_col_names = list(relig_pivot.columns)
+   relig_pivot.columns = ["A", "B", "C", "D"]
+   relig_pivot.dropna(how="all", inplace=True)
+
+.. note::
+
+   In the variable ``religiosity_col_names`` we have stored the names of the values in order. This implies that
+
+   A = Extremely important
+   B = Not at all important
+   C = Not very important
+   D = Somewhat important
+
+To perform ANOVA, we are going to use a Python library called "stats", in particular the f_oneway function. You can find more information about this library by typing ``stats`` in the Help Pane. And if you want to learn more about the function we are going to use, type ``stats.f_oneway`` in the Help pane:
+
+.. image:: images/workshop-2/help-pane-for-stats.f_oneway.png
+   :alt: stats.f_oneway() help in Spyder's Help pane
+
+The help notes that this function requires that you pass, as an array-like, the measurements for each group (in our case, one sample for A, one for B, one for C, and one for D). Since we have empty cells in each column, let's remove them from each column (the f_oneway function does not support missing values).
```

Comment:
```suggestion
The help notes that this function requires an array-like with the measurements for each group as an input (in our case, one sample for A, one for B, one for C, and one for D). Since we have empty cells in each column, let's remove them from each column (the ``f_oneway`` function does not support missing values).
```